### PR TITLE
 Show warning dialog before disabling the backups

### PR DESCRIPTION
**Pull request by nyarian**
tari-project/wallet-android#475

### CODE REVIEW

File: app/src/main/java/com/tari/android/wallet/infrastructure/backup/BackupManager.kt [hunk 1]
```diff
@@ -241,6 +241,7 @@ internal class BackupManager(
             /* no-op */
             Logger.e(
                 exception,
+                // TODO(nyarian): but why ignoring this?
```
Comment (kukabi) [inline app/src/main/java/com/tari/android/wallet/infrastructure/backup/BackupManager.kt:244]: It's a non-user-actionable state, and if backups were turned on all other backup files will be deleted on every new backup. So it's not a blocker, can be ignored safely.

Comment (nyarian) [reply app/src/main/java/com/tari/android/wallet/infrastructure/backup/BackupManager.kt:244]: backups can basically be not deleted if network fails or permissions are revoked, is it?

Comment (nyarian) [reply app/src/main/java/com/tari/android/wallet/infrastructure/backup/BackupManager.kt:244]: it seems like a continuation of "atomic ui" concept from the other side - like backups are not enabled if it fails to make the first backup, then it's the same here: backups are not disabled if it failed to delete them

Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/infrastructure/backup/BackupManager.kt:244]: If auth is revoked we can assume that backup is turned off. As for the network connection case, I think the user shouldn't have to have internet connection to be able to tell the system that they don't want it to backup anymore...yet...if we fail to delete the backup file then they could restore from it - this shouldn't be possible when backup was turned-off previously.

There's also the possibility of revoking access but keeping the files - not sure if the files are accessible later when the app gets authed again.

So let's keep the `TODO`s and reconsider later 👍  We should go for the simplest solution in general imho.

**Pull request merged**